content/typeclasses-syntax.md

```diff
@@ -0,0 +1,380 @@
+---
+layout: sip
+stage: implementation
+status: implemented
+presip-thread: https://contributors.scala-lang.org/t/pre-sip-improve-syntax-for-context-bounds-and-givens/6576/97
+title: Improve Syntax for Context Bounds and Givens
+---
+
+**By: Martin Odersky**
+
+## History
+
+| Date          | Version            |
+|---------------|--------------------|
+| March 11, 2024| Initial Draft      |
+
+## Summary
+
+We propose some syntactic improvements that makes context bounds and given clauses more
+expressive and easier to read. The proposed additions and changes comprise:
+
+ - naming context bounds, as in `A: Monoid as a`,
+ - a new syntax for multiple context bounds, as in `A: {Monoid, Ord}`,
+ - context bounds for type members,
+ - a cleaner syntax for given definitions that eliminates some syntactic warts,
+ - replacing abstract givens with a more powerful and convenient mechanism.
+
+## Motivation
+
+This SIP is part of an effort to get state-of-the art typeclasses and generic in Scala. It fixes several existing pain points:
+
+ - The inability to name context points causes awkward and obscure workarounds in practice.
+ - The syntax for multiple context bounds is not very clear or readable.
+ - The existing syntax for givens is unfortunate, which hinders learning and adoption.
+ - Abstract givens are hard to specify and implement and their syntax is easily confused
+   with simple concrete givens.
+
+These pain points are worth fixing on their own, independently of any other proposed improvements to typeclass support. What's more, the changes
+are time sensitive since they affect existing syntax that was introduced in 3.0, so it's better to make the change at a time when not that much code using the new syntax is written yet.
+
+
+ 1. Named tuples are a convenient lightweight way to return multiple results from a function. But the absence of names obscures their meaning, and makes decomposition with _1, _2 ugly and hard to read. The existing alternative is to define a class instead. This does name fields, but is more heavy-weight, both in terms of notation and generated bytecode. Named tuples give the same convenience of definition as regular tuples at far better readability.
```

Comment:
Is this from this SIP?
  It's unclear how named tuples are connected with context bounds and givens.

Reply:
Oops, no that was an oversight. Deleted.